### PR TITLE
ThreddsCatalogGroup and a few WebMapServiceCatalogItem fixes

### PR DESCRIPTION
**Pull request by rowanwins**
### What this PR does

- [x] This PR creates a `ThreddsCatalogGroup` model. THREDDS is a server used for scientific data, often with many many rasters exposed as WMS's.

````
{
      "type": "thredds-group",
      "url": "http://dapds00.nci.org.au/thredds/catalog/rr9/ASCAT/ASCAT_v1-0_soil-moisture_daily_0-05deg_2007-2011/00000000/catalog.xml",
      "name": "thredds example"
 }
````

- [x] Fixes a small bug on WMS GetFeatureInfo to ensure `styles`  parameter is not undefined by making it an empty string.
- [x] Ports `supportsColorScaleRange`, `colorScaleMinimum` and `colorScaleMaximimum` from `master`, this enables rescaling of WMS presentation which is important with THREDDS.

### Checklist

-   [x] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
-   [x] I've updated CHANGES.md with what I changed.

### CODE REVIEW

File: lib/Models/WebMapServiceCatalogItem.ts [hunk 1]
```diff
@@ -459,6 +465,19 @@ class GetCapabilitiesStratum extends LoadableStratum(
     }
   }
 
+  @computed
+  get isThredds(): boolean {
+    if (this.catalogItem.url) {
+      if (this.catalogItem.url.indexOf("thredds") > -1) return true;
```
Comment (rowanwins) [inline lib/Models/WebMapServiceCatalogItem.ts]: This probably isn't the best check although I've never seen a thredds server without `thredds` in the title

File: lib/Models/WebMapServiceCatalogItem.ts [hunk 2]
```diff
@@ -883,7 +913,8 @@ class WebMapServiceCatalogItem
         "width",
         "height",
         "bbox",
-        "layers"
+        "layers",
+        "version"
```
Comment (rowanwins) [inline lib/Models/WebMapServiceCatalogItem.ts:1016]: This is here as a temporary fix until Cesium implements this fix
https://github.com/CesiumGS/cesium/issues/9021

I should probably annotate a To-DO to remove it in the future

File: lib/Traits/WebMapServiceCatalogItemTraits.ts [hunk 3]
```diff
@@ -259,4 +259,35 @@ export default class WebMapServiceCatalogItemTraits extends mixTraits(
       "Gets or sets the coverage name for linked WCS for clip-and-ship."
   })
   linkedWcsCoverage?: string;
+
+  @primitiveTrait({
+    type: "string",
+    name: "Is Thredds",
+    description: "True if this WMS is from a THREDDS server; otherwise, false."
+  })
+  isThredds: boolean = false;
```
Comment (rowanwins) [inline lib/Traits/WebMapServiceCatalogItemTraits.ts:275]: I always get confused whether things should be optional if they have a default, eg 
````
  isThredds?: boolean = false;
````

File: lib/ReactViews/FeatureInfo/FeatureInfoSection.jsx [hunk 4]
```diff
@@ -254,6 +257,42 @@ export const FeatureInfoSection = observer(
         this.renderDataTitle();
       const reactInfo = getInfoAsReactComponent(this);
 
+      let threddsChart = null;
```
Comment (rowanwins) [inline lib/ReactViews/FeatureInfo/FeatureInfoSection.jsx]: This is the section of the PR that most concerns me. It generates a pixel drill chart in the feature info that is pretty specific to THREDDS.

File: lib/Models/ThreddsCatalogGroup.ts [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+    );
+
+    threddsGroup.setTrait(CommonStrata.definition, "name", catalog.title);
+    threddsGroup.setTrait(CommonStrata.definition, "url", catalog.url);
+    threddsGroup.terria.addModel(threddsGroup);
+  }
+
+  @action
+  async createMembers() {
+    this.threddsCatalog = await threddsCrawler(this._catalogGroup.url);
```
Comment (nf-s) [inline lib/Models/ThreddsCatalogGroup.ts]: We need to be able to proxy this url, but the URL parser inside the thredds-catalog-crawler doesn't like our proxy URLs:

For example:  
`proxy/_1d/http://climate-data.it.csiro.au:8080/thredds/catalog/data/authoritative/VIC-5/catalog.xml`

```bash
2urlParsingBrowser.js?a9a8:2 Uncaught (in promise) TypeError: Failed to construct 'URL': Invalid URL
    at Requestor.parseUrl (urlParsingBrowser.js?a9a8:2)
    at new Catalog (Catalog.js?08ae:19)
    at threddsCatalogCrawler (main.js?918f:4)
    at entry (entryBrowser.js?b652:13)
    at ThreddsStratum._callee$ (ThreddsCatalogGroup.ts?2f16:103)
    at tryCatch (runtime.js?96cf:63)
    at Generator.invoke [as _invoke] (runtime.js?96cf:293)
    at Generator.eval [as next] (runtime.js?96cf:118)
    at asyncGeneratorStep (ThreddsCatalogGroup.ts?2f16:18)
    at _next (ThreddsCatalogGroup.ts?2f16:18)
```

**Review by nf-s — CHANGES_REQUESTED** [review]

File: lib/Models/ThreddsCatalogGroup.ts (hunk 5 above)
Comment (nf-s) [reply lib/Models/ThreddsCatalogGroup.ts]: I'm happy to take this on tomorrow :)

**Comment by soyarsauce** [comment]
@nf-s to pick this up

**Comment by soyarsauce** [comment]
For CSIROMap

File: lib/Models/ThreddsCatalogGroup.ts (hunk 5 above)
Comment (nf-s) [reply lib/Models/ThreddsCatalogGroup.ts]: fixed

File: lib/Models/WebMapServiceCatalogItem.ts (hunk 2 above)
Comment (nf-s) [reply lib/Models/WebMapServiceCatalogItem.ts:1016]: added todo

**Review by nf-s — APPROVED** [review]

**Review by nf-s — APPROVED** [review]

File: lib/ReactViews/FeatureInfo/FeatureInfoSection.jsx (hunk 4 above)
Comment (nf-s) [reply lib/ReactViews/FeatureInfo/FeatureInfoSection.jsx]: no longer relevant

**Pull request merged**